internal/controller/merge.go

```diff
@@ -160,10 +160,18 @@ func mergeIncomingFilters(curr, toMerge []frr.IncomingFilter) []frr.IncomingFilt
 	mergedIn := map[string]*frr.IncomingFilter{}
 	for _, a := range all {
 		f := a
-		_, found := mergedIn[f.Prefix]
-		if found {
+		curr, found := mergedIn[f.Prefix]
+		if !found {
+			mergedIn[f.Prefix] = &f
 			continue
 		}
+
+		if f.LEThan == 0 || curr.LEThan > f.LEThan {
+			f.LEThan = curr.LEThan
+		}
+		if f.GEThan == 0 || curr.GEThan < f.GEThan {
```

Comment:
shouldn't this be `curr.GE > 0 && curr.GE < f.GE` otherwise we always merge GE to 0 (when there's a prefix without ge present)?